app/main.py

```diff
@@ -1,2 +1,124 @@
+from dataclasses import dataclass
+from fractions import Fraction
+from typing import Hashable, Any, Iterator, Optional
+
+
 class Dictionary:
-    pass
+    @dataclass
+    class Node:
+        key: Hashable
+        obj_hash: hash
+        value: Any
+
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.size = 0
+        self.hash_table: list[Dictionary.Node | None] = [None] * self.capacity
+
+    def _calculate_index(self, key: Hashable) -> int:
+        index = hash(key) % self.capacity
+
+        while (
+            self.hash_table[index]
+            and self.hash_table[index].key != key
+        ):
+            index = (index + 1) % self.capacity
+
+        return index
+
+    @property
+    def current_max_size(self) -> Fraction:
+        return self.capacity * Fraction(2 , 3)
+
+    def _resize(self) -> None:
+        old_hash_table = self.hash_table
+        self.capacity *= 2
+        self.size = 0
+        self.hash_table = [None] * self.capacity
+        for node in old_hash_table:
+            if node:
+                self[node.key] = node.value
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.size > self.current_max_size:
+            self._resize()
+        index = self._calculate_index(key)
+        if not self.hash_table[index]:
+            self.size += 1
+        self.hash_table[index] = Dictionary.Node(key, hash(key), value)
+
+    def __getitem__(self, item: Any) -> Any:
+        index = self._calculate_index(item)
+
+        if self.hash_table[index] is None:
+            raise KeyError(f"Cannot find object with key -> {item}")
+
+        return self.hash_table[index].value
+
+    def __delitem__(self, key: Hashable) -> None:
+        index = self._calculate_index(key)
+
+        if self.hash_table[index] is None:
+            raise KeyError(f"Cannot find value for key: {key}")
+
+        self.hash_table[index] = None
+        self.size -= 1
+
+    def __len__(self) -> int:
+        return self.size
+
+    def __iter__(self) -> Iterator:
+        return iter(self.hash_table)
+
+    def get(self, key: Hashable) -> Hashable | None:
+        try:
+            return self[key]
+        except KeyError:
+            print(f"Error! Value with key {key} does not exist")
+        return None
+
+    def clear(self) -> None:
+        self.__init__()
+
+    def pop(self, key: Hashable, default: Optional[Any]) -> Any | None:
+        temp = None
+        index = self._calculate_index(key)
+        if self.hash_table[index]:
```

Comment:
you can just `try` to get an element using brackets interface and then do something else in except clause.
Also keep in mind that pop method should have this behaviour:
Dictionary.pop(key, None) should return None instead of triggering KeyError
Dictionary.pop(key) should raise KeyError